docs/okeps/okep-5094-layer2-transit-router.md

```diff
@@ -0,0 +1,991 @@
+    stor-GR-node1 <--> stor-transit_router
+    stor-GR-node2 <--> rtos-GR-node2
+    stor-GR-node1-node2 <--> stor-GR-node2
+
+    class VM vmStyle;
+    class rtotr-GR-node1 portStyle;
+    class rtos-GR-node2 portStyle;
+    class stor-GR-node2 portStyle;
+    class stor-GR-node1-node2 portStyle;
+    class stor-GR-node1 portStyle;
+    class trtor-GR-node1 portStyle;
+    class trtor-GR-node2 portStyle;
+    class stor-transit_router portStyle;
+    class rtos-layer2-switch portStyle;
+    class GR-node1 routerStyle;
+    class GR-node2 routerStyle;
+    class transit_router_node1 routerStyle;
+    class layer2-switch_node1 switchStyle;
+    class layer2-switch_node2 switchStyle;
+    class term termStyle;
+    class node1,node2 nodeStyle;
+```
+
+### Egress IP changes (TBD)
+
+### Transit subnet conflict
+
+Before this change Layer2 networks using Subnet overlapping with the `transitSubnet` were allowed to be created (and would be working just fine).
+Now it is not possible anymore, and we need to decide what to do in this case:
+1. Before upgrading to the new topology, check that no Layer2 networks with overlapping subnet are present.
+If they are, block the upgrade and inform the user to fix the issue. This will require one more release of supporting the old topology
+(and in the next release we break the networks if they were not upgraded)
+2. Simply upgrade all networks, and let the overlapping UDNs break, informing the user to fix the issue afterwards.
+
+### API Details
+
+Every node will get a temporary annotation "k8s.ovn.org/layer2-topology-version: 2.0" that will be removed in the next
+ovn-k version (1.3 is the feature gets in 1.2).
+Every Layer2 and Layer3 NAD will get new annotations for tunnel keys distribution.
+
+### Implementation Details
+
+A [PoC PR](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5067) exists with basic implementation to make all CI pass and as input
+for this document.
+
+#### At ovnkube-node
+
+Create a helper that is able to derive the peer IPs from the node annotation
+for the transit peers node subnet.
+
+The layer2 controller should do the following:
+- Adapt and call the syncNodeClusterRouterPort from layer3 that creates the LRP that connects to the switch at transit_router
+- Pass transit_router as cluster_name to the gateway init functions
+- Remove from gateway.go the code that was attaching GR to layer2 switch
+- Add to gateway.go the code that connects GR to transit_router using the peer ports
+- Change egressip.go so routes are configured at transit_router instead of gateway router.
+- Change the ZoneInterConnect handler to be able to add the transit router remote ports to transit_router
+- In general re-use as much Layer3 code as possible since these makes Layer2 topology similar to it.
+
+#### At ovnkube-control-plane
+
+At cluster manager at the zoneCluster controller calculate the peers node subnet
+and annotate the node.
+
+To calculate the peer node subnet we have two options:
```

Comment:
this is the first time in the enhancement there is a reference to "peer node subnet". I'm guessing this is transit-peers-node-subnet. Can we clarify that here?